examples/77_blackwell_fmha/collective/fmha_fusion.hpp

```diff
@@ -157,7 +157,8 @@ struct CausalMask : NoMask {
       TileShape const& tile_shape,
       ProblemSize const& problem_size) {
 
-    return ceil_div(get<0>(tile_shape), get<1>(tile_shape));
+    int trip_count = get_trip_count(blk_coord, tile_shape, problem_size);
+    return std::min(trip_count, int(ceil_div(get<0>(tile_shape), get<1>(tile_shape))));
```

Comment:
```suggestion
    return std::min(trip_count, int(ceil_div(size<0>(tile_shape), size<1>(tile_shape))));
```

do not assume tile shapes are depth 0

Reply:
But all Mask-related classes (NoMask, ResidualMask, CausalMask) in this file assume depth-0.
Do you want me to change all uses?

Reply:
let's start small and at least make this change here locally.